jaxlib/xla/sharding.cc

```diff
@@ -223,9 +223,22 @@ NamedSharding::NamedSharding(nb::object mesh, nb::object spec,
   // TODO(phawkins): this leaks a reference to the check_pspec function.
   // A better way to fix this would be to move PartitionSpec and this check into
   // C++.
-  static nb::object* check_pspec = []() {
+  static nb::object* check_pspec = [](){
```

Comment:
You must not put "static" here. Otherwise you just have the same deadlock we had before.

Reply:
Thank you pointing out this! Definitely I had the same deadlock as before.